src/coreclr/vm/siginfo.cpp

```diff
@@ -4551,7 +4554,8 @@ BOOL MetaSig::CompareFieldSigs(
     pEndSig1 = pSig1 + cSig1;
     pEndSig2 = pSig2 + cSig2;
 
-    CompareState state{ pVisited };
+    TokenPairList visited { pVisited };
+    CompareState state{ &visited };
     return(CompareElementType(++pSig1, ++pSig2, pEndSig1, pEndSig2, pModule1, pModule2, NULL, NULL, &state));
```

Comment:
Is the `state` ever `NULL` now? The `CompareElementType()`'s last argument is optionally `NULL`. Can we instead make it a reference to avoid all of this confusion?

Reply:
We still have cases where `pVisited` is `NULL` and I think we need to be able to update the token list in place, so I think we need to keep it as a pointer.